drivers/serial/uart_nrfx_uart.c

```diff
@@ -1145,18 +1173,44 @@ static void uart_nrfx_pins_enable(const struct device *dev, bool enable)
 static int uart_nrfx_pm_action(const struct device *dev,
 			       enum pm_device_action action)
 {
+	const struct uart_nrfx_config *config = get_dev_config(dev);
```

Comment:
When `CONFIG_PINCTRL` is not defined, an "unused variable" warning may be generated for `config`. I think you could pass this variable as a parameter to `uart_nrfx_pins_configure` to prevent the warning (this would save `#ifdef CONFIG_PINCTRL` here and in `uart_nrfx_init()`).

Reply:
I prefer to use `dev` as an argument, so guarded config using ifdef.